From 7553ce81f721ee6d01879bb81f2f650b22c52ad7 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Mon, 8 Sep 2014 07:30:43 -0700
Subject: [PATCH] unpack.c: fix for new libarchive

Fixed:
* The archive_read_support_compression_all() isn't needed any more.
* archive_read_finish() -> archive_read_free()
* ARCHIVE_DEFAULT_BYTES_PER_BLOCK isn't available any more, use 10240.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 loader/unpack.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/loader/unpack.c b/loader/unpack.c
index f7a5777..c83bd31 100644
--- a/loader/unpack.c
+++ b/loader/unpack.c
@@ -60,9 +60,6 @@ int unpack_init(struct archive **a) {
     if ((*a = archive_read_new()) == NULL)
         return ARCHIVE_FATAL;
 
-    if ((r = archive_read_support_compression_all(*a)) != ARCHIVE_OK)
-        return r;
-
     if ((r = archive_read_support_format_all(*a)) != ARCHIVE_OK)
         return r;
 
@@ -120,7 +117,7 @@ int unpack_members_and_finish(struct archive *a, char *dest,
         return ARCHIVE_FATAL;
     }
 
-    if (archive_read_finish(a) != ARCHIVE_OK) {
+    if (archive_read_free(a) != ARCHIVE_OK) {
         logMessage(ERROR, "error closing archive (%s:%d): %s",
                    __func__, __LINE__, archive_error_string(a));
         return ARCHIVE_FATAL;
@@ -152,8 +149,7 @@ int unpack_archive_file(char *filename, char *dest) {
         return rc;
     }
 
-    rc = archive_read_open_filename(a, filename,
-                                    ARCHIVE_DEFAULT_BYTES_PER_BLOCK);
+    rc = archive_read_open_filename(a, filename, 10240);
     if (rc != ARCHIVE_OK) {
         logMessage(ERROR, "error opening %s (%s:%d): %s",
                    filename, __func__, __LINE__,
-- 
1.7.9.5

